enhancements/automatic-resource-pruning.md

```diff
@@ -0,0 +1,196 @@
+primary entry point function that takes in configuration and prunes resources accordingly. The configuration will accept
+one or many resource types, a pruning strategy, namespaces, label selectors, and other common settings such as a dry run
+mode, hooks, and logging configuration.
+
+Another aspect of the library will be determining when it can and cannot prune a resource. For example, the prune
+functionality should not remove a running `Pod` until it has completed, even if it meets the strategy criteria. The
+library will expose a way for operator authors to specify what criteria makes a resource of a specific type safe to
+prune; this can include checking annotations, a status, or any other data available on the resource. An important
+distinction to draw is the difference between a strategy and checking whether it is safe to prune a resource (henceforth
+called the "is-pruneable" functionality). Strategies look generically at collections of resources and decide which
+resources in the collection to prune, if any. They should only take criteria common to all Kubernetes resources into
+account such as the count of resources and creation timestamp. The is-pruneable functionality conversely looks at one
+and only one resource at a given point in time to determine whether or not the current prune task can remove a resource
+based on its specific data.
+
+Note that stategies will not be programmatically limited to being resource agnostic, but it will be a defined best
+practice to write strategies in such a way. One exception to this recommendation will be when the operator author wants
+to prune based on a cumulative value such as the summation of a field across multiple resources. The operator author
+must then be sure to add safe guards to the strategy to avoid unexpected behavior if used with an incompatible resource.
+
+### User Stories
+
+#### Story 1
+
+An an operator author, I want to limit the number of `Jobs` in a completed state on the cluster so that the long term
+storage cost of my operator has a cap.
+
+#### Story 2
+
+As an operator author, I want to limit the number of `Pods` in a completed state on a the cluster and preserve the
+`Pods` in an error state so that the long term storage cost of my operator has a soft cap and the user can see status
+information about failed `Pods` before manually removing.
+
+#### Story 3
+
+As an operator author, I want to limit the number of `Pods`with a custom heuristic based on the creation timestamp so
+that the long term storage cost of my operator has a cap based on my operator's logic.
+
+#### Story 4
+
+As an operator author, I want to prune a custom resources with specific status information when there is a certain
+number so that the long term storage cost of my operator has a cap.
+
+#### Story 5
+
+As an operator author, I want to prune both `Jobs` and `Pods` of a certain age so that the long term storage cost of my
+operator has a cap and there are no orphaned resources.
+
```

Comment:
As an additional user story I would mention that as an operator author I may want to be able to inject my own custom strategy for determining which resource is eligible for pruning and which resource to prune